rct229/rulesets/ashrae9012019/section19/section19rule21.py

```diff
@@ -0,0 +1,510 @@
+            "4. Heating systems in Climate Zones 0 through 3."
+            "5. Cooling systems in Climate Zones 3C, 4C, 5B, 5C, 6B, 7, and 8."
+            "6. Where the largest exhaust source is less than 75% of the design outdoor airflow. This exception shall only be used if exhaust air energy recovery is not used in the proposed design."
+            "7. Systems requiring dehumidification that employ energy recovery in series with the cooling coil. This exception shall only be used if exhaust air energy recovery and series-style energy recovery coils are not used in the proposed design.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1.2.10 and exceptions 1-7",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_instances/0",
+            list_path="$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+            data_items={"climate_zone": ("baseline", "weather/climate_zone")},
+        )
+
+    def is_applicable(self, context, data=None):
+        climate_zone = data["climate_zone"]
+
+        return climate_zone in APPLICABLE_CZ_ZONES
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        rmi_p = context.proposed
+        climate_zone = rmi_b["weather"]["climate_zone"]
+
+        ER_not_req_for_heating_sys = (
+            True if climate_zone in APPLICABLE_CZ_ZONES else False
+        )
+
+        dict_of_zones_and_terminal_units_served_by_hvac_sys = (
+            get_dict_of_zones_and_terminal_units_served_by_hvac_sys(rmi_b)
+        )
+
+        hvac_sys_and_assoc_zones_largest_exhaust_source = {}
+        for hvac_id_b in find_all(
+            "$.buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+            rmi_b,
+        ):
+            for zone_id_b in dict_of_zones_and_terminal_units_served_by_hvac_sys[
+                hvac_id_b
+            ]["zone_list"]:
+                space_p = find_one(
+                    f"$.buildings[*].building_segments[*].zones[?(@.id == {zone_id_b})].spaces",
+                    rmi_p,
+                )
+                if space_p.get("lighting_space_type") is not None:
+                    lighting_space_type_p = space_p["lighting_space_type"]
+                    serves_zones_with_systems_likely_exhausting_toxic_etc = (
+                        True
+                        if lighting_space_type_p
+                        == LIGHTING_SPACE.LABORATORY_EXCEPT_IN_OR_AS_A_CLASSROOM
+                        else False
+                    )
+                    serves_kitchen_space = (
+                        True
+                        if lighting_space_type_p == LIGHTING_SPACE.FOOD_PREPARATION_AREA
+                        else False
+                    )
```

Comment:
I think both parameters here serves_zones_with_systems_likely_exhausting_toxic_etc and serves_kitchen_space should just have a true case:
```python
if lighting_space_type_p == LIGHTING_SPACE.LABORATORY_EXCEPT_IN_OR_AS_A_CLASSROOM:
    serves_zones_with_systems_likely_exhausting_toxic_etc = True
```
Otherwise, these two flags could be flipped multiple times in a loop.

Reply:
good point! Addressed.